controllers/secret_controller.go

```diff
@@ -79,12 +73,18 @@ func (r *SecretReconciler) SetupWithManager(mgr ctrl.Manager) error {
 		Complete(r)
 }
 
-func (r *SecretReconciler) getAuthConfigsUsingAPIKey(ctx context.Context) authConfigSet {
+func (r *SecretReconciler) eachAuthConfigsWithK8sSecretBasedIdentity(ctx context.Context, f func(*evaluators.AuthConfig)) {
```

Comment:
We don't need passing the `ctx context.Context` in here, do we?
Unclear to me what our stance here is/should be. I've seen a few patterns in the code (first arg, last arg, pass it despite not being used, not pass it, ...)

Reply:
Well spotted! I think I left this one behind in the cleanup. I was using it for debugging logs added temporarily in dev, that later didn't make sense to keep in the final version.

Reply:
Fixed